subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/impl/SynchronizationReport.kt

```diff
@@ -0,0 +1,219 @@
+package com.avito.android.test.report.impl
+
+import com.avito.android.test.report.InternalReport
+import com.avito.android.test.report.ReportState
+import com.avito.android.test.report.StepException
+import com.avito.android.test.report.model.DataSet
+import com.avito.android.test.report.model.StepResult
+import com.avito.android.test.report.model.TestMetadata
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.locks.ReentrantLock
+
+internal class SynchronizationReport(
+    private val report: InternalReport
+) : InternalReport {
+    private val instrumentationThread = AtomicReference<Thread?>()
```

Comment:
Hold a reference to the instrumentation thread because some operations must be done only from that thread